src/backend/distributed/planner/recursive_planning.c

```diff
@@ -916,15 +916,6 @@ ShouldRecursivelyPlanSubquery(Query *subquery, RecursivePlanningContext *context
 		 */
 		return false;
 	}
-	else if (TaskExecutorType == MULTI_EXECUTOR_TASK_TRACKER &&
-			 SingleRelationRepartitionSubquery(subquery))
-	{
-		/*
-		 * Citus can plan this and execute via repartitioning. Thus,
-		 * no need to recursively plan.
-		 */
-		return false;
```

Comment:
wonder if it's worth preserving this (e.g. via a separate GUC)

Reply:
We should probably remove the SingleRelationRepartitionSubquery logic fully, or keep it.

Reply:
And do you have a preference? I don't have enough context to decide which one is better

Reply:
I added this logic back.